lib/friendly_shipping/services/usps_ship/api_error.rb

```diff
@@ -0,0 +1,29 @@
+# frozen_string_literal: true
+
+require 'friendly_shipping/api_error'
+
+module FriendlyShipping
+  module Services
+    class USPSShip
+      class ApiError < FriendlyShipping::ApiError
+        # @param [RestClient::Exception] cause
+        def initialize(cause)
+          super(cause, parse_message(cause))
+        end
+
+        private
+
+        # @param [RestClient::Exception] error
+        # @return [String]
+        def parse_message(error)
+          return error.message unless error.response
+
+          parsed_json = JSON.parse(error.response.body)
+          parsed_json.dig("error", "message")
+        rescue JSON::ParserError, KeyError => _e
```

Comment:
nit: how would a KeyError get generated?

Reply:
Good point. Since I'm not using `fetch` anywhere, it wouldn't. I'll remove.